src/usr/lib/rsetup/cli/system.sh

```diff
@@ -83,6 +84,21 @@ update_emmc_boot() {
     done
 }
 
+update_generic_hostname() {
+    __parameter_count_range_check 1 4 "$@"
+
+    local current_hostname
+    current_hostname=$(cat < /etc/hostname | tr -d '\n')
+
+    for arg in "$@"
+    do
+        if [ "$current_hostname" == "$arg" ]
```

Comment:
```suggestion
    if __in_array "$current_hostname" "$@" >/dev/null
```
我们有提供判断元素是否在数组里的helper